### PR TITLE
fix node label deletion

### PR DESCRIPTION
**Pull request by AviAvni**
fixes https://github.com/RedisGraph/RedisGraph/issues/2357
https://github.com/RedisGraph/RedisGraph/issues/2361

### CODE REVIEW

File: tests/unit/test_graph.cpp [hunk 1]
```diff
@@ -395,16 +395,11 @@ TEST_F(GraphTest, RemoveNodes) {
 	uint edge_count = array_len(edges);
 	ASSERT_EQ(edge_count, 2);
 
-	// Delete the edges on the first node manually, as Graph_DeleteNode expects
-	// it to be detached.
-	// Both 0 -> 1 edge and 1 -> 0 edge should be removed.
-	for(uint i = 0; i < edge_count; i ++) Graph_DeleteEdge(g, &edges[i]);
+	Graph_GetNode(g, 0, &node);
+	Graph_BulkDelete(g, &node, 1, edges, edge_count, NULL, NULL);
```
Comment (swilly22) [inline tests/unit/test_graph.cpp:399]: This test function `RemoveNodes` doesn't release the lock `Graph_AcquireWriteLock`

File: tests/unit/test_graph.cpp [hunk 2]
```diff
@@ -986,7 +981,10 @@ TEST_F(GraphTest, GraphStatistics) {
 	ASSERT_EQ(edge_deleted, 3);
 
 	// delete disconnected node
-	Graph_DeleteNode(g, &n[3]);
+	Graph_AcquireWriteLock(g);
```
Comment (swilly22) [inline tests/unit/test_graph.cpp:984]: lock is acquired at line 971, do we really need to acquire and release multiple times in this function ?

File: tests/flow/test_multi_label.py [hunk 3]
```diff
@@ -229,3 +229,33 @@ def test09_test_query_graph_populate_nodes_labels(self):
...
+
+        query = """MATCH (a) DELETE a"""
+        query_result = graph.query(query)
+        self.env.assertEquals(query_result.nodes_deleted, 3)
+
+        query = """CREATE (a:L4) RETURN labels(a)"""
+        query_result = graph.query(query)
+        self.env.assertEquals(query_result.labels_added, 1)
+        self.env.assertEquals(query_result.nodes_created, 1)
+        self.env.assertEquals(query_result.result_set[0][0], ["L4"])
```
Comment (swilly22) [inline tests/flow/test_multi_label.py:261]: I'll assume without the fix this `a` node would have 2 labels ?

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: tests/flow/test_multi_label.py (hunk 3 above)
Comment (AviAvni) [reply tests/flow/test_multi_label.py:261]: yes

**Review by swilly22 — APPROVED** [review]

File: tests/flow/test_multi_label.py (hunk 3 above)
Comment (Rudy0530) [reply tests/flow/test_multi_label.py:261]: Yes, and there may be more than one extra label.  

I have encountered a situation where a node is marked with three or even four labels

**Pull request merged**